core/src/main/java/org/bitcoinj/script/ScriptBuilder.java

```diff
@@ -177,6 +177,8 @@ protected ScriptBuilder bigNum(int index, long num) {
 
         if (num == 0) {
             data = new byte[0];
+        } else if (num == -1) {
```

Comment:
Hmm, this change violates the contract from the method comment, which guarantees to always use pushdata (hence the "big"). I think the fix needs to go into the two number() methods.